services/src/main/java/org/broadleafcommerce/payment/service/gateway/DefaultPayPalCheckoutTransactionService.java

```diff
@@ -243,19 +262,19 @@ public PaymentResponse authorizeAndCapture(PaymentRequest paymentRequest) {
 
     @Override
     public PaymentResponse reverseAuthorize(PaymentRequest paymentRequest) {
-        PaymentResponse paymentResponse =
-                new PaymentResponse(DefaultPaymentTypes.THIRD_PARTY_ACCOUNT,
-                        PayPalCheckoutPaymentGatewayType.PAYPAL_CHECKOUT)
-                                .transactionType(DefaultTransactionTypes.REVERSE_AUTH);
+        PaymentResponse paymentResponse = paymentResponseUtil.buildPaymentResponse(paymentRequest,
+                PayPalCheckoutPaymentGatewayType.PAYPAL_CHECKOUT,
+                DefaultTransactionTypes.REVERSE_AUTH);
```

Comment:
We should move the `paymentResponseUtil.buildPaymentResponse(...)` call inside of the `try` block & we should catch `Exception` instead of `PaymentException`, to guarantee that this method will always return a PaymentResponse